openedx/core/djangoapps/course_views/course_views.py

```diff
@@ -0,0 +1,176 @@
+"""
+Tabs for courseware.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+from xmodule.tabs import CourseTab, link_reverse_func
+
+_ = lambda text: text
+
+
+# Stevedore extension point namespaces
+COURSE_VIEW_TYPE_NAMESPACE = 'openedx.course_view_type'
+
+
+class CourseViewType(object):
+    """
+    Base class of all course view type plugins.
```

Comment:
@cpennington I had a great discussion with Ned (who I won't tag so he doesn't get spammed). His thoughts are:
* this feature has always been called "tabs" so he prefers that over "views" which he found too confusing. If mobile shows course tabs using a different UX metaphor then that is okay.
* he also finds "type" to be too confusing in the Python sense. He first thought of "factory" since this thing creates tabs, but it also does more. His next suggestion was "descriptor" which we both liked a lot.

I'm now leaning towards ``CourseTabDescriptor`` with the entry point called ``course_tab``. @cpennington, let me know if you like this, or if you have further suggestions. I'd like to do the renaming Tuesday morning if possible so we can merge this story by Wednesday.